pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -640,15 +640,20 @@ private static String database(Properties props) {
    * @return the timeout from the URL, in milliseconds
    */
   private static long timeout(Properties props) {
+    int result = 0;
     String timeout = PGProperty.LOGIN_TIMEOUT.get(props);
     if (timeout != null) {
       try {
-        return (long) (Float.parseFloat(timeout) * 1000);
+        result = Integer.parseInt(timeout);
       } catch (NumberFormatException e) {
         LOGGER.log(Level.WARNING, "Couldn't parse loginTimeout value: {0}", timeout);
       }
     }
-    return (long) DriverManager.getLoginTimeout() * 1000;
+    if (result == 0) {
```

Comment:
Pease use `org.postgresql.PGProperty#isPresent` to check property presence.
`result != 0` is not a proper check as user might specify `loginTimeout=0` explicitly to disable login timeout.